### ISSUE TITLE
LibJS: Correct length for emoji (and other unicode) strings

### ISSUE THREAD

**Issue by linusg** [SerenityOS/serenity#2280]
Lots of fun ahead.

Currently:

```js
"⛳".length === 3
"🔥".length === 4
"🔥🔥🔥".length === 12
"👨‍👩‍👦".length === 18 // asserts when pasting into the REPL
"👩‍❤️‍💋‍👩".length === 27 // asserts when pasting into the REPL
```

Goal:

```js
"⛳".length === 1
"🔥".length === 2
"🔥🔥🔥".length === 6
"👨‍👩‍👦".length === 8
"👩‍❤️‍💋‍👩".length === 11
```

**Comment by alimpfard** [comment]
Following the unicode-ification of libline, the copy-paste should not assert anymore.
but LibJS (or the js repl) will have to report the indices in codepoints now, which is...not really ideal tbh

**Comment by linusg** [comment]
> Following the unicode-ification of libline, the copy-paste should not assert anymore.

It's hitting this `ASSERT_NOT_REACHED()`:

https://github.com/SerenityOS/serenity/blob/3bc3f36cfe820472779e53b9b20ddcd7647e5d2f/Libraries/LibLine/Editor.cpp#L830-L836

So I assume that'll be fixed in https://github.com/SerenityOS/serenity/pull/2281 (where it's being removed).

**Comment by alimpfard** [comment]
ohhhh.
:thinking: why is it hitting that assert? everything pasted should be valid utf-8...

**Comment by Zenyiar** [comment]
I know this is random but linusg, I know Andreas dosen't like me or my Feature Requests but it would be nice if you could right click on elements on the browser and click inspect element and the inspect element menu show up and auto select the element.

**Comment by awesomekling** [comment]
JavaScript is specced to use UTF-16 strings, so this will be a bit cumbersome indeed. :)

A common solution in other browser engines is to use UTF-16 as the native string type throughout the engine.

**Comment by MineNewsX** [comment]
This is bullshit, i read the code of conduct and have no reason to be blocked from SerenityOS

**Comment by awesomekling** [comment]
> This is bullshit, i read the code of conduct and have no reason to be blocked from SerenityOS

https://github.com/SerenityOS/serenity/blob/master/CODE_OF_CONDUCT.md
"Nobody owes you anything."

You have been spamming this issue tracker with feature requests and untested non-working patches for a while now, and repeatedly been told to stop. Since you're not listening, I'm blocking you from interacting with the repository.

**Comment by linusg** [comment]
Fixed in #8897.
